### DISCUSSION TITLE
[Deta Base] returns empty for query on the large data

### DISCUSSION THREAD

**Discussion by nghiepdev** [deta/cloud-docs#686] [Help] [answered]
Hi,

It is likely that the Deta Base query is no longer accurate when the data is large enough.

https://user-images.githubusercontent.com/4768095/209569525-48f61e53-ba9c-4c6f-a400-6b6c516724a8.mov

**Comment by etrotta** [comment, marked as answer]
Quoting the [documentation](https://docs.deta.sh/docs/base/sdk#fetch):

> Up to 1 MB of data is retrieved before filtering with the query. Thus, in some cases you might get an empty list of items but still the last key evaluated in the response.
> 
> To apply the query through all the items in your base, you have to call fetch until last is empty.

when you query for a key, it can find it directly
when you query with a filter for a field like the 'secret', it has to iterate though the entire thing
